CMakePresets.json

```diff
@@ -13,35 +13,40 @@
         "PRESET_IN_USE": "1"
       }
     },
-    {
-      "name": "cross",
-      "displayName": "STM32 G4 OT-3 cross-compilation",
-      "description": "Build application firmware for OT-3 systems that use STM32, for flashing onto boards",
+    {"name": "cross-no-directory-reqs",
+      "displayName": "STM32 G4 OT-3 cross-compilation with no directory specifications",
+      "description": "Build application firmware for OT-3 systems that use STM32",
       "generator": "Unix Makefiles",
-      "binaryDir": "${sourceDir}/build-cross",
```

Comment:
Why don't we need the binary directory anymore to build hex files?

Reply:
We do, but we want a way to build the firmware where we can specify the binary directory.
The `cross` preset is behaviorally unchanged - you still don't have to specify anything. What this does is add a new preset where you have to specify the build dir with `-B`. That's really specifically and only (but very) useful in openembedded, where by convention you build stuff somewhere completely unrelated from the source directory. This lets OE specify an arbitrary directory to build in while keeping the dev experience unchanged.